### PR TITLE
refactor(card): use `billing.first_name` instead of `card_holder_name`

### PR DESCRIPTION
**Pull request by Narayanbhat166**
## Type of Change

- [x] Refactoring

## Description

This PR includes changes to use `billing.first_name` and `billing.last_name` instead of `payment_method_data.[card].card_holder_name` at the connector level. The backwards compatibility is provided.

So `billing.first_name` will be used instead of `card_holder_name` in the connectors.

### Additional Changes

- [x] This PR modifies the API contract

## How did you test it?

- Added unit tests
- Create a payment of stripe connector, and check the data in stripe dashboard.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [x] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/api_models/src/payments.rs [hunk 1]
```diff
@@ -4696,15 +4753,18 @@ mod billing_from_payment_method_data {
...
         let card_payment_method_data = PaymentMethodData::Card(Card {
-            card_holder_name: Some(Secret::new(TEST_FIRST_NAME.into())),
+            card_holder_name: Some(Secret::new(TEST_FIRST_NAME_SINGLE.into())),
             ..Default::default()
         });
 
-        let billing_address = card_payment_method_data.get_billing_address().unwrap();
+        let billing_address = card_payment_method_data.get_billing_address();
+
+        dbg!(&billing_address);
```
Comment (hrithikesh026) [inline crates/api_models/src/payments.rs:4762]:
```suggestion
```

File: crates/router/src/connector/utils.rs [hunk 2]
```diff
@@ -353,6 +355,12 @@ impl<Flow, Request, Response> RouterData for types::RouterData<Flow, Request, Re
             .ok_or_else(missing_field_err("recurring_mandate_payment_data"))
     }
 
+    fn get_optional_billing_combined_name(&self) -> Option<Secret<String>> {
```
Comment (hrithikesh026) [inline crates/router/src/connector/utils.rs:358]: why `combined_name`? why not `full_name`?

File: crates/router/src/connector/threedsecureio/transformers.rs [hunk 3]
```diff
@@ -328,6 +328,18 @@ impl TryFrom<&ThreedsecureioRouterData<&types::authentication::ConnectorAuthenti
             })?;
         let meta: ThreeDSecureIoConnectorMetaData =
             to_connector_meta(request.pre_authentication_data.connector_metadata.clone())?;
+
+        let card_holder_name = billing_address
```
Comment (ArjunKarthik) [inline crates/router/src/connector/threedsecureio/transformers.rs:332]: Why can't we use get_optional_billing_full_name here?

**Review by ThisIsMani — DISMISSED** [review]
Verify connector changes looks fine.

**Review by ArjunKarthik — APPROVED** [review]

**Review by hrithikesh026 — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by ThisIsMani — APPROVED** [review]

**Review by ThisIsMani — APPROVED** [review]

**Pull request merged**